File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        
        ArrayList<Integer> rank = new ArrayList<>();
        
        for(int i = 0; i < score.length; i++) {
            if(i < k) {
                rank.add(score[i]);
                if(rank.size() >= 2) rank.sort((o1, o2) -> o1 - o2);
            }
            else {
                if(score[i] > rank.get(0)) {
                    rank.remove(0);
                    rank.add(score[i]);
                    rank.sort((o1, o2) -> o1 - o2);
                }
            }
            
            answer[i] = rank.get(0);
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []

    stack = []

    for sitem in score:
        if len(stack) < k:
            stack.append(sitem)
        else:
            if sitem > stack[-1]:
                stack.pop()
                stack.append(sitem)    

        stack.sort(reverse=True)   

        answer.append(stack[-1]) 

    return answer
